bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/MessageBox.java

```diff
@@ -199,7 +266,16 @@ public int open () {
 	TCHAR buffer1 = new TCHAR (0, message, true);
 	TCHAR buffer2 = new TCHAR (0, title, true);
 	display.externalEventLoop = true;
+
+	/* Create the callback hook */
+	Callback cbtCallback = null;
+	if (labels != null) {
+		cbtCallback = new Callback (this, "CBTProc", 3); //$NON-NLS-1$
+		cbtHook = OS.SetWindowsHookEx (OS.WH_CBT, cbtCallback.getAddress (), 0, OS.GetCurrentThreadId ());
+	}
+
 	int code = OS.MessageBox (hwndOwner, buffer1, buffer2, bits);
+	if (cbtCallback != null) cbtCallback.dispose();
```

Comment:
Just for the extra safety, I would `OS.UnhookWindowsHookEx()` here (just before disposing callback) as well if it wasn't uninstalled before. Otherwise, the hook might remain installed if something goes contrary to expected (for example for whatever reason `HCBT_ACTIVATE` never arrives). This would in turn cause a JVM crash when it tries to call a disposed callback.

Reply:
Done.